server/services/caldav/lib/calendar/calendar.syncUserCalendars.js

```diff
@@ -53,7 +53,10 @@ async function syncUserCalendars(userId) {
 
       // Else update it if sync is enable on calendar & events change
       if (gladysCalendar[0].sync && formatedCalendar.ctag !== gladysCalendar[0].ctag) {
-        await this.gladys.calendar.update(gladysCalendar[0].selector, formatedCalendar);
+        await this.gladys.calendar.update(gladysCalendar[0].selector, {
+          ...formatedCalendar,
+          sync: gladysCalendar[0].sync,
```

Comment:
Don't you risk to still update wrongly if it changed between the .get and the .update ?

Why not simply removing the sync attribute from the update if it's not required?

Reply:
There is no action between get and update, so time is really, really short, I don't think change can happen. But indeed delete is a better option, more readable I changed.